src/ORM/Query.php

```diff
@@ -287,7 +287,7 @@ public function eagerLoader(EagerLoader $instance = null)
      */
     public function contain($associations = null, $override = false)
     {
-        if (empty($associations) && $override) {
+        if ($override) {
             $this->_eagerLoader = null;
```

Comment:
Is there a way to separate containments that came from matching() / notMatching() once they've been added to the eagerloader?